### ISSUE TITLE
RF builder custom templates: Surprise (fuzzy?) translation happening

### ISSUE THREAD

**Issue by jennyhavah** [mercycorps/TolaActivity#2322]
I discovered this in the course of creating a custom template. It appears that we are automatically translating custom RF tiers, which is unexpected, and problematic in that it results in a mishmash of languages for languages other than the one in which the template was created.

![image](https://user-images.githubusercontent.com/33670923/79796237-10069580-830a-11ea-86ae-8cf7bfb27658.png)

To reproduce:

1. Create a custom RF template that includes a mix of unique tiers and tiers used in our standard templates such as Goal or Activity.

2. Add at least one indicator to the Goal and/or Activity tier.

3. Go to the program page.

4. Note that these words are translated, whereas the unique tier names are not.

**Comment by sanjuroj** [comment]
Did you check anywhere else for this (e.g. IPTT or program page?)  It may be happening there too.  I suspect this is just the general behavior of the translation machinery, that it's seeing a word on its list of things to translate and translating it.  This may not bite us all that often.  I'm guessing that if a user has set their language to something other than English, that they'll be entering non-English values for the tier names.  This isn't related to fuzzy translations, those are guesses that the system makes when it sees new translations.  

All that said, I _think_ this is going to be a heavy lift to fix. There is a context mechanism that's meant to help with a somewhat similar problem but it doesn't really help us here.  As an example, when we added the descriptive text about number displays to the profile, the word "Period" was used differently than we'd used it in, say, periodic targets.  I added a context tag to the profile page that told the translation machinery to translate Period differently there than elsewhere.  But here we are differentiating how we want to translate in the exact same context.  To do this, I think we'll need to ensure we pass the template type to wherever these tier names are displayed (which is a lot of places now) and put a conditional around the display to say whether it should be translated or not.  Checking other parts of the application to see if it's happening there will help confirm whether or not this problem is widespread.  If it's not, I may be overestimating the LoE.  I'd also really like to see what @camgnostic has to say about it.

**Comment by jennyhavah** [comment]
@sanjuroj You're right that it's everywhere, and must be a function of the translation machinery. 

Clarification: My understanding was that a string needs to have some form of translation tags around it, for it to be picked up by the translation machinery. So I'm wondering: Do we have translation tags around the custom RF tiers, and if so, why? (I'm sure it's more complicated than this, but in my mind, it'd be a matter of removing those tags.)

**Comment by sanjuroj** [comment]
For those displays, the app is currently blind to what type of tier name it is.  It can't tell custom tags from standard ones, it just is told "here's a tier name, try to translate it".  So we'd have to add that ability to differentiate to every place it's displayed.

**Comment by jennyhavah** [comment]
Thanks for the clarification, @sanjuroj. 

@camgnostic what do you think? Is this a high level of effort?
